fail hard if stage 2 fails to create directory

jan--f opened the pull request:
Fixes: #81

theanalyst commented on srv/modules/runners/push.py:
```diff
@@ -72,6 +72,14 @@ def proposal(filename = "/srv/pillar/ceph/proposals/policy.cfg", dryrun = False)
+        os.makedirs(path)
+    except OSError as err:
+        if err.errno == errno.EACCES:
+            print "ERROR: Cannot create dir {}".format(path)
```
does the print statement also get displayed when raising the exception ? (or do we also need a log.error? )

jan--f replied:
It does get printed. But I agree...I should call log here.

rjfd commented:
@jan--f please squash the first and third commits.

jschmid1 commented on srv/modules/runners/push.py:
```diff
@@ -72,6 +72,16 @@ def proposal(filename = "/srv/pillar/ceph/proposals/policy.cfg", dryrun = False)
     pillar_data.output(common)
     return True
 
+def __create_dirs(path, root):
```
since we have the exact same method here and in `populate.py:757` we might consider to create a helper lib.

jan--f replied:
I'd be all for that. Just haven't found an answer where such a lib would live. '''/srv/modules/runners''' sounds like a bad idea to me, since salt would pick this up as a runner.

jan--f replied:
@swiftgist Any idea what the convention would be, if there is any?

swiftgist replied:
I haven't found it yet.  I suspect we could have something live under /usr/lib/python2.7/site-packages/salt, but that does not help with packaging.  I suppose the closest I have come with a common function for runners is a module.  This might seem strange or abusive, but we have more than this create_dirs example that is common across runners.

swiftgist commented:
Double check the behavior of the orchestration file for Stage 2.  I had to rely on the side effect of printing in other cases from runners since the orchestration will not capture the error effectively and only gives a stack trace.  I have also been avoiding stack traces since finding the errors are difficult.

jan--f commented:
hmm it does print ok-ish. The runner still doesn't fail as I'd like. What do you think @swiftgist ?
```
[ERROR   ] 
            ERROR: Cannot create dir /srv/pillar/ceph/proposals/cluster-ceph/cluster
            Please make sure /srv/pillar/ceph/proposals is owned by salt
            
Traceback (most recent call last):
  File "/srv/modules/runners/populate.py", line 759, in _create_dirs
    os.makedirs(path)
  File "/usr/lib64/python2.7/os.py", line 150, in makedirs
    makedirs(head, mode)
  File "/usr/lib64/python2.7/os.py", line 157, in makedirs
    mkdir(name, mode)
OSError: [Errno 13] Permission denied: '/srv/pillar/ceph/proposals/cluster-ceph'
retcode:
    0
admin_master:
  Name: minions.ready - Function: salt.runner - Result: Changed Started: - 15:01:49.692495 Duration: 538.227 ms
  Name: populate.proposals - Function: salt.runner - Result: Changed Started: - 15:01:50.230955 Duration: 804.119 ms

Summary for admin_master
------------
Succeeded: 2 (changed=2)
Failed:    0
------------
Total states run:     2
Total run time:   1.342 s
```

swiftgist commented:
That seems fine.  The error message is at the top and the stack trace isn't too long to cause things to scroll off.  Maybe log.exception appears more prominent.  If that's the case, we should make the same change in other runners.

jan--f commented:
@jschmid1 should we defer the code duplication until we have a good solution for this?
